target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -440,7 +447,7 @@ fn post_price_update_from_vaa<'info>(
 
     match message {
         Message::PriceFeedMessage(price_feed_message) => {
-            price_update_account.write_authority = payer.key();
+            price_update_account.write_authority = write_authority.key();
```

Comment:
can you also save the slot when this update was written to the account here?

Reply:
I recommend doing it on the Oracle Instance.

Reply:
Done